frontend/src/libs/components/avatar/avatar.tsx

```diff
@@ -0,0 +1,26 @@
+import { Link } from '~/libs/components/link/link.jsx';
+import { AppRoute } from '~/libs/enums/app-route.enum';
+import { getFirstLetters } from '~/libs/helpers/helpers.js';
+
+import styles from './styles.module.scss';
+
+type Properties = {
+  username: string;
+  avatarUrl?: string | null;
+};
+
+const Avatar: React.FC<Properties> = ({ avatarUrl, username }) => {
+  return (
+    <Link to={AppRoute.PROFILE} className={styles['avatar-info']}>
+      {avatarUrl ? (
+        <img src={avatarUrl} alt="avatar" className={styles.avatar} />
+      ) : (
+        <span className={`${styles.avatar} ${styles.defaultAvatar}`}>
+          <span className={styles.initials}>{getFirstLetters(username)}</span>
+        </span>
+      )}
+    </Link>
+  );
```

Comment:
1 - link should not be in common Avatar url - other places won't need a link when using Avatar component -- wrap Avatar component with link in Header (where you need it)
2 - let's use early return for avatar placeholder
3 - there is getValidClassNames helper to join multiple classes assignment to nodes
```suggestion
  if (!avatarUrl) {
    return (
      <span className={getValidClassNames(styles.avatar, styles.defaultAvatar)}`}>
        <span className={styles.initials}>{getFirstLetters(username)}</span>
      </span>
    );
  }

  return <img src={avatarUrl} alt="avatar" className={styles.avatar} />;
```